index.js

```diff
@@ -3,13 +3,26 @@ const path = require('path');
 const resolveFrom = require('resolve-from');
 const callerPath = require('caller-path');
 
-const clear = moduleId => {
+const clear = (moduleId, options = {}) => {
 	if (typeof moduleId !== 'string') {
 		throw new TypeError(`Expected a \`string\`, got \`${typeof moduleId}\``);
 	}
 
+	if (typeof options !== 'object') {
+		throw new TypeError(`Expected an \`object\`, got \`${typeof options}\``);
+	}
+
 	const filePath = resolveFrom(path.dirname(callerPath()), moduleId);
 
+	// Delete its children from module
+	if (options.children && require.cache[filePath] && require.cache[filePath].children) {
+		let i = require.cache[filePath].children.length;
+
+		while (i--) {
+			delete require.cache[require.cache[filePath].children[i].id];
+		}
+	}
```

Comment:
Would it maybe be smart to run the children through `clear()`, making it recursive, instead of deleting them directly. That way the children too would be run through the code on line 26-35 that deletes the children from their parents, preventing memory leaks.

Reply:
Oh, I was just working on a recursive implementation before I notice your comment.

Reply:
Mm, yes, it probably would have been easier to use `clear()` and comparing root directory to make sure files are nested inside.